### DISCUSSION TITLE
Concepts on BaggingForecaster

### DISCUSSION THREAD

**Discussion by BradKML** [alan-turing-institute/sktime#3573] [Q&A]
In reference to [BaggingForecaster](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.compose.BaggingForecaster.html) and [EnsembleForecaster](https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.forecasting.compose.EnsembleForecaster.html):
1. Does the `BaggingForecaster` and `EnsembleForecaster` include hyperparameter optimization, or does not naively fit each sub-model?
2. If not, how can this be combined with `ForecastingGridSearchCV` by bagging the top N forecasters?
3. How are these two different, as they both seem to be similar in structure, and that the former provides confidence interval while the latter does not? small note: "ensemble" has been used to either mean "bagging" and boosting (as in gradient boost) interchangeably, which make it a bit confusing

**Comment by topher-lo** [comment]
@BrandonKMLee 
1. If you are refering to automatically identifying the best weights for each forecaster, have you looked at [AutoEnsembleForecaster](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.compose.AutoEnsembleForecaster.html). If you mean are the parameters of the underlying forecasters gridsearched, the answer is: it depends on the forecaster (e.g. AutoARIMA or AutoETS).
2. You'll have to clarify what hyperparameters you are referring to
3. A lot of points are covered here that is hard to untangle. But I think your question suggests a decent point. R fable for forecasting makes a distinction between ensemble and combination forecasting, which sktime seems to conflate (https://youtu.be/nMy98aqQj8c?t=1192)

@fkiraly thoughts? https://youtu.be/nMy98aqQj8c?t=1192
If you watch the video above, it seems we should consider separating ensemble and combination forecasters. Both forecasters also give different forecast distributions, which we can also very easily compute quantiles from.

**Reply by topher-lo** [reply]
@RNKuhns @GuzalBulatova @aiwalter @mloning Pinging authors.

**Reply by BradKML** [reply]
For 2 `ForecastingGridSearchCV` to find top N forecasters with their respective weights and parameters, then mixing "best weights for each forecaster" with `BaggingForecaster` or `EnsembleForecaster`. This should hopefully decrease overfitting?

For 3 the example is slightly clearer but it might be useful to have it documented through API doc or Tutorials.
